### PR TITLE
Only use parse PHP files when building stub cache

### PR DESCRIPTION
**Pull request by dantleech**
... and test improvements

### CODE REVIEW

File: tests/Integration/Core/SourceCodeLocator/StubSourceLocatorTest.php [hunk 1]
```diff
@@ -13,32 +14,54 @@
...
-        $code = $this->cacheBuilder->locate(ClassName::fromString('StubOne'));
+        $this->workspace()->put('stubs/Stub.php', '<?php class StubOne {}');
+        $code = $this->sourceLocator->locate(ClassName::fromString('StubOne'));
         $this->assertContains('class StubOne', (string) $code);
     }
 
     public function testCanLocateFunctions()
     {
-        $code = $this->cacheBuilder->locate(Name::fromString('hello_world'));
+        file_put_contents($this->workspace()->path('stubs/Stub.php'), '<?php function hello_world() {}');
```
Comment (dantleech) [inline tests/Integration/Core/SourceCodeLocator/StubSourceLocatorTest.php]: Use `->put()`n

File: lib/Core/SourceCodeLocator/StubSourceLocator.php [hunk 2]
```diff
@@ -62,7 +62,7 @@ private function buildCache()
     {
         $map = [];
         foreach ($this->fileIterator() as $file) {
-            if ($file->isDir()) {
+            if ($file->getExtension() !== 'php' || $file->isDir()) {
```
Comment (Ocramius) [inline lib/Core/SourceCodeLocator/StubSourceLocator.php]: Note: `.phtml` probably also needed

Comment (dantleech) [reply lib/Core/SourceCodeLocator/StubSourceLocator.php]: This is a stub source locator - would stubs ever need `.phtml`? I think a more general indexing locator would need it however.

Comment (Ocramius) [reply lib/Core/SourceCodeLocator/StubSourceLocator.php]: Ah, probably ok as-is then 👌

**Pull request merged**